Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM golang:1.10.1-alpine3.7 as go-builder
+
+ARG PROJECT_NAME=zookeeper-operator
+ARG REPO_PATH=github.com/pravega/$PROJECT_NAME
+ARG BUILD_PATH=${REPO_PATH}/cmd/${PROJECT_NAME}
+
+# Build version and commit should be passed in when performing docker build
+ARG VERSION=0.0.0-localdev
+ARG GIT_SHA=0000000
+
+COPY pkg /go/src/${REPO_PATH}/pkg
+COPY cmd /go/src/${REPO_PATH}/cmd
+COPY vendor /go/src/${REPO_PATH}/vendor
+
+RUN GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o ${GOBIN}/${PROJECT_NAME} \
+    -ldflags "-X ${REPO_PATH}/pkg/version.Version=${VERSION} -X ${REPO_PATH}/pkg/version.GitSHA=${GIT_SHA}" \
+    $BUILD_PATH
+
+# =============================================================================
+FROM alpine:3.7 AS final
```

Comment:
Can you explain why we have two `FROM` here. I do not understand the problem we are solving/how it works. Can you explain in the PR comment/comment in the file what we are trying to do here?

Reply:
It's called a multi-stage build. The idea behind is that we use the first stage to configure and build our application, including all the necessary tooling and dependencies. Then, the second stage normally uses a lightweight base image (e.g. `alpine`) and copies the binary from the first stage, keeping the final image clean and light. You can read more about it here: https://docs.docker.com/develop/develop-images/multistage-build/

Reply:
Thanks.